uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/absence-period-instance.js

```diff
@@ -0,0 +1,50 @@
+define([
+  'leave-absences/shared/modules/models-instances',
+  'common/moment',
+  'common/models/instances/instance',
+], function (instances, moment) {
+  'use strict';
+
+  instances.factory('AbsencePeriodInstance', ['$log', 'ModelInstance',
+    function ($log, ModelInstance) {
+      $log.debug('AbsencePeriodInstance');
+
+      return ModelInstance.extend({
+        /**
+         * Sets the current property of this absence period on instantiation
+         * and then creates a new instance.
+         * As its overriding the ModelInstance.init method it needs to include isInPeriod
```

Comment:
I think I understand what you're saying here...but then wouldn't it be better to amend `ModelInstance` to have a method called `customInit` or something, that `init()` will always call, and that children instances can define if needed? Something like:
```js
instances.factory('ModelInstance', function () {

  return {
    
    /**
      * Method that a children instance can redefine to perform custom initialization logic
      */
    customInit: _.noop,

    init: function () {
      // ...
      
      this.customInit();

      return _.assign(Object.create(this), attributes);
    }
  }
```
```js
instances.factory('AbsencePeriodInstance', function () {
    return ModelInstance.extend({
      defaultCustomData: function () {
        return { current: false; }
      }

      customInit: function () {
        this.current = // perform your check
      }
    });
});
```
Do you think this would work?